### PR TITLE
[RFR] Isolate admin sub domains by adding globalized selectors

### PR DESCRIPTION
**Pull request by djhi**
- [x] Add globalized selectors for each subdomain
- [x] Ensure with refer to the fields sub domain each time we needs fields
- [x] Publication sub domain responsability reduced to check whether data has been published

### CODE REVIEW

File: src/app/js/admin/index.js [hunk 1]
```diff
@@ -17,5 +17,12 @@ export default combineReducers({
     publish: publishReducer,
     removedResources,
     upload,
-    validation,
 });
+
+export const fromFields = createGlobalSelectors(s => s.admin.fields, fieldsSelectors);
```
Comment (ThieryMichel) [inline src/app/js/admin/index.js]: should be moved in an AdminSelectors files.

File: src/app/js/admin/fields/FieldForm.js [hunk 2]
```diff
@@ -14,8 +14,7 @@ import {
     saveField,
 } from './';
 
-import { fromFields } from '../../selectors';
-
+import { fromFields } from '../';
```
Comment (ThieryMichel) [inline src/app/js/admin/fields/FieldForm.js]: What do we import here?
When we import an index we sort of import a directory, and a directory should not be imported from its children IMO. avoid all from '../'

File: src/app/js/admin/fields/TransformerListItem.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ import ActionDeleteIcon from 'material-ui/svg-icons/action/delete';
 import { Field, FieldArray, propTypes as reduxFormPropTypes } from 'redux-form';
 
 import { polyglot as polyglotPropTypes } from '../../propTypes';
-import { getTransformers, getTransformerArgs } from './';
+import { fromFields } from '../';
```
Comment (ThieryMichel) [inline src/app/js/admin/fields/TransformerListItem.js]: same

File: src/app/js/admin/fields/sagas/addField.js [hunk 4]
```diff
@@ -5,7 +5,9 @@ import {
     addFieldError,
     addFieldSuccess,
 } from '../';
-import { fromFields } from '../../../selectors';
+
+import { fromFields } from '../../';
```
Comment (ThieryMichel) [inline src/app/js/admin/fields/sagas/addField.js]: same

File: src/app/js/admin/fields/sagas/addField.spec.js [hunk 5]
```diff
@@ -7,11 +7,10 @@ import {
     addFieldError,
     addFieldSuccess,
 } from '../';
+import { fromFields } from '../../';
```
Comment (ThieryMichel) [inline src/app/js/admin/fields/sagas/addField.spec.js]: same

File: src/app/js/admin/fields/sagas/prepareTransformers.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import { call, select } from 'redux-saga/effects';
 
 import {
-    getTransformerArgs,
-} from '../';
+    fromFields,
+} from '../..';
```
Comment (ThieryMichel) [inline src/app/js/admin/fields/sagas/prepareTransformers.js]: same

File: src/app/js/admin/fields/sagas/prepareTransformers.spec.js [hunk 7]
```diff
@@ -2,8 +2,8 @@ import expect from 'expect';
 import { call, select } from 'redux-saga/effects';
 
 import {
-    getTransformerArgs,
-} from '../';
+    fromFields,
+} from '../../';
```
Comment (ThieryMichel) [inline src/app/js/admin/fields/sagas/prepareTransformers.spec.js]: same

File: src/app/js/admin/fields/sagas/validation.js [hunk 8]
```diff
@@ -1,17 +1,15 @@
...
     LOAD_FIELD_SUCCESS,
+    setValidation,
     ADD_FIELD_SUCCESS,
     REMOVE_FIELD_SUCCESS,
     UPDATE_FIELD_SUCCESS,
-} from '../fields';
-import { fromFields } from '../../selectors';
+} from '../';
+
+import { fromFields } from '../../';
```
Comment (ThieryMichel) [inline src/app/js/admin/fields/sagas/validation.js]: same

File: src/app/js/admin/parsing/ParsingResult.js [hunk 9]
```diff
@@ -10,12 +10,11 @@ import Card from '../../lib/Card';
...
 import {
-    getParsedExcerptColumns,
-    getExcerptLines,
-    isParsingLoading,
-    getTotalLoadedLines,
     clearParsing,
 } from './';
+import {
+    fromParsing,
+} from '../';
```
Comment (ThieryMichel) [inline src/app/js/admin/parsing/ParsingResult.js]: same

File: src/app/js/admin/publicationPreview/sagas.js [hunk 10]
```diff
@@ -17,9 +17,10 @@ import {
     UPDATE_FIELD_SUCCESS,
 } from '../fields';
 
+import { fromFields, fromParsing } from '../';
```
Comment (ThieryMichel) [inline src/app/js/admin/publicationPreview/sagas.js]: same

File: src/app/js/admin/publish/Publish.js [hunk 11]
```diff
@@ -8,17 +8,15 @@ import FlatButton from 'material-ui/FlatButton';
 import { polyglot as polyglotPropTypes } from '../../propTypes';
 
 import {
-    getPublish,
     publish as publishAction,
 } from './';
-import { getIsPublished } from '../publication';
-import { getIsValid } from '../validation';
+import { fromFields, fromPublish, fromPublication } from '../';
```
Comment (ThieryMichel) [inline src/app/js/admin/publish/Publish.js]: same

File: src/app/js/admin/publish/Validation.js [hunk 12]
```diff
@@ -1,6 +1,6 @@
 import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
-import { getInvalidFields } from '../validation';
+import { fromFields } from '../';
```
Comment (ThieryMichel) [inline src/app/js/admin/publish/Validation.js]: same

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: src/app/js/admin/fields/FieldForm.js (hunk 2 above)
Comment (djhi) [reply src/app/js/admin/fields/FieldForm.js]: No, standard node when not specifying a file is to import `index`

File: src/app/js/admin/index.js (hunk 1 above)
Comment (djhi) [reply src/app/js/admin/index.js]: I wanted to avoid yet another file

File: src/app/js/admin/fields/FieldForm.js (hunk 2 above)
Comment (ThieryMichel) [reply src/app/js/admin/fields/FieldForm.js]: Yes, and index is the index of the directory, it should be used to import the module represented by the directory, it is weird for a module to import itself.

**Pull request merged**